core-api/src/main/java/com/optimizely/ab/bucketing/Bucketer.java

```diff
@@ -102,6 +109,19 @@ private Variation bucketToVariation(@Nonnull Experiment experiment,
                                         @Nonnull String userId) {
         // "salt" the bucket id using the experiment id
         String combinedBucketId = userId + experiment.getId();
+
+        // If a persistent bucketer instance is present then check it for a persisted variation
+        if (userExperimentRecord != null) {
+            String variationKey = userExperimentRecord.lookup(userId, experiment.getKey());
+            if (variationKey != null) {
+                logger.info("Returning previously activated variation \"{}\" from user experiment registry.", variationKey);
+                // A variation is stored for this combined bucket id
+                return projectConfig.getExperimentIdMapping().get(experiment.getId()).getVariationKeyToVariationMap().get(variationKey);
+            } else {
+                logger.info("No previously activated variation returned from persistent bucketer.");
+            }
+        }
+
         String experimentKey = experiment.getKey();
```

Comment:
move this above the newly added block to reuse `experimentKey` rather than calling `experiment.getKey()` again on line 115